src/ASM/ASMs3Dmx.C

```diff
@@ -175,9 +175,20 @@ bool ASMs3Dmx::generateFEMTopology ()
 {
   if (!svol) return false;
 
-  if (m_basis.empty())
+  if (m_basis.empty()) {
     m_basis = ASMmxBase::establishBases(svol, ASMmxBase::Type);
 
+    // we need to project on something that is not one of our bases
+    if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
+        ASMmxBase::Type == ASMmxBase::DIV_COMPATIBLE) {
+      auto vec2 = ASMmxBase::establishBases(svol,
+                                            ASMmxBase::FULL_CONT_RAISE_BASIS1);
+      projBasis = vec2.front();
+    }
```

Comment:
Same here, can avoid the `vec2` temporary, I think

Reply:
done.

Reply:
removed.